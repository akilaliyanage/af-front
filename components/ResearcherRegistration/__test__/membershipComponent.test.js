import React  from 'react';
import ReactDOM from 'react-dom';
import {render, screen, cleanup} from '@testing-library/react'
import MembershipComponent from '../membershipComponent'
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer'
import '@testing-library/jest-dom/extend-expect'
import expectExport from 'expect';

//clean up
afterEach(()=>cleanup())


//Test for successful rendering of the membership componant
it('Chech membership succssful rendering',() =>{
    const div = document.createElement('div')
    ReactDOM.render(
            <BrowserRouter>
                <MembershipComponent/>
            </BrowserRouter>
    ,div)

})

//Test for successful rendering of the membership componant - contain the label
it('membership Componant renders correctly', () =>{
    render(
            <BrowserRouter>
                <MembershipComponent />
            </BrowserRouter>
    )
    const memberContainer = screen.getByTestId('membership-container')
    expectExport(memberContainer).toContainHTML("</label>")
})


//Compare snapshot of membershipcomponanent dor testing
it('snapshot check for memberships', () =>{
    const componant = renderer.create(
            <BrowserRouter>
                <MembershipComponent/>
            </BrowserRouter>
    ).toJSON();

    expect(componant).toMatchSnapshot()
    
})
