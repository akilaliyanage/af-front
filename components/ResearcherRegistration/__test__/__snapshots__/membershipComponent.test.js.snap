// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot check for memberships 1`] = `
<div
  className="form-row-container"
  data-testid="membership-container"
>
  <div
    className="form-Element-container"
  >
    <input
      className="Researcher-form-input-long"
      id="membership"
      name="membership"
      type="text"
    />
    <label>
      Name of Membership
    </label>
  </div>
</div>
`;
